engine-tests/src/main/java/org/terasology/engine/integrationenvironment/jupiter/IntegrationEnvironment.java

```diff
@@ -13,5 +15,34 @@
 @Target(ElementType.TYPE)
 @Retention(RetentionPolicy.RUNTIME)
 public @interface IntegrationEnvironment {
+    /**
+     * The network mode the host engine starts with.
+     * <p>
+     * See {@link NetworkMode} for details on the options.
+     * <p>
+     * Some modes automatically include a {@link LocalPlayer}.
+     * <p>
+     * If you want to simulate multiple players with
+     * {@link org.terasology.engine.integrationenvironment.Engines#createClient Engines.createClient},
+     * you will need a mode with a {@linkplain NetworkMode#isServer() server}.
+     */
     NetworkMode networkMode() default NetworkMode.NONE;
+
+    /**
+     * Add an additional subsystem to the engine.
+     * <p>
+     * A new instance will be included in the engine's subsystems when it is created.
+     * <p>
+     * Implementing {@link EngineSubsystem#initialise} gives you the opportunity to
+     * make changes to the configuration <em>before</em> it would otherwise be available.
+     */
+    Class<? extends EngineSubsystem> subsystem() default NO_SUBSYSTEM.class;
+
+    /**
+     * Do not add an extra subsystem to the integration environment.
+     * <p>
+     * [Odd marker interface because annotation fields cannot default to null.]
```

Comment:
:question: Can/should one use `Optional`? Can there be more than subsystem added (in principle, not necessarily in practice)? If so, this could also be a list. 

In any case, having this marker interface in `engine-tests` is just fine, as it is definitely understandable and we should probably not spent too much time and effort on this (now).

Reply:
I don't think we can. The list of types allowed to annotations is [quite limited](https://docs.oracle.com/javase/specs/jls/se11/html/jls-9.html#jls-9.6.1).